src/chi-vue/src/components/data-table-toolbar/DataTableToolbar.vue

```diff
@@ -1,5 +1,12 @@
 <script lang="ts" setup>
 import { DATA_TABLE_CLASSES } from '@/constants/classes';
+import ColumnCustomization from '../../components/column-customization/ColumnCustomization';
+import { DataTableColumnsData } from '@/constants/types';
+
+const props = defineProps<{
+  columnCustomizationBtn?: boolean;
+  columnsData?: DataTableColumnsData | {};
```

Comment:
We can avoid creating two props and check if there is columns data. Also, columnsData cannot be `{}` as it requires the key `columns`:

```suggestion
  columnCustomizationData?: DataTableColumnsData;
```

```vue
        <ColumnCustomization v-if="columnCustomizationData" :columnsData="columnCustomizationData" />
```